ftw/zipextract/events.py

```diff
@@ -0,0 +1,8 @@
+from interfaces import IFile
+from interfaces import IZip
+from Products.Five.utilities.marker import mark
+
+
+def set_zip_on_creation(file, event):
+    if IFile(file)._is_zip():
+        mark(file, IZip)
```

Comment:
I'm not yet happy with the approach of marking ZIPs.
One disadvantage is that we then need to process all files in an upgrade and apply the interface.

I'd prefer to not have this interface.
If it is only for the bumblebee action, I'd test the mimetype directly in the view.

Reply:
Got rid of the marker interface. I'm simply testing the mimetype when generating the bumblebee actions. I'm also testing the mimetype before extracting from a zip (in case someone calls the action directly via URL).